src/serialization/sb3.js

```diff
@@ -25,6 +25,87 @@ const {deserializeCostume, deserializeSound} = require('./deserialize-assets.js'
  * @property {Map.<string, string>} extensionURLs - map of ID => URL from project metadata. May not match extensionIDs.
  */
 
+const serializeBlock = function (block) {
+    const obj = Object.create(null);
+    obj.id = block.id;
+    obj.opcode = block.opcode;
+    obj.next = block.next;
+    obj.parent = block.parent;
+    obj.inputs = block.inputs;
+    obj.fields = block.fields;
+    obj.topLevel = block.topLevel ? block.topLevel : false;
+    obj.shadow = block.shadow;
+    if (block.topLevel) {
+        if (block.x) {
+            obj.x = Math.round(block.x);
+        }
+        if (block.y) {
+            obj.y = Math.round(block.y);
+        }
+    }
+    if (block.mutation) {
+        obj.mutation = block.mutation;
+    }
+    return obj;
+};
+
+const serializeBlocks = function (blocks) {
+    const obj = Object.create(null);
+    for (const blockID in blocks) {
+        obj[blockID] = serializeBlock(blocks[blockID]);
```

Comment:
In `for...in` loops, we should use [hasOwnProperty](https://github.com/LLK/scratch-vm/blob/develop/src/blocks/scratch3_sound.js#L236) to make sure we're only looping on instance-level properties and not those added to the prototype (https://stackoverflow.com/questions/12735778/for-in-and-hasownproperty).

Reply:
Would it be better to just use `for...of` and Object.keys?

```js
for (const blockID of Object.keys(blocks)) {
  ...
}
```